post_processing/ci_plots.jl

```diff
@@ -98,17 +154,66 @@ function parse_var_attributes(var)
     return join(info, ", ")
 end
 
+"""
+    make_plots_generic(
+        output_path::Union{<:AbstractString, Vector{<:AbstractString}},
+        vars,
+        args...;
+        plot_fn = nothing,
+        output_name = "summary",
+        summary_files = String[],
+        MAX_NUM_COLS = 1,
+        MAX_NUM_ROWS = min(4, length(vars)),
+        kwargs...,
+    )
+
+Use `plot_fn` to plot `vars` properly handling pagination.
+
+Arguments
+=========
+
+`output_path` can be a `String` or a list of `String`s. When it is a list of `String`s, it
+is assumed that `vars` are coming from different simulations and they have to be compared.
+Hence, the summary plot is saved to the first `output_path` and the number of columns is
+fixed to be the same as the number of `output_path`s. `summary_files` are also assumed to be
+in the first `output_path`.
+
+`output_name` is the name of the file produced.
+
+`summary_files` is an optional list of paths to prepend to the PDF produced by this
+function. This is useful when building larger and more complex reports that required
+different `plot_fn` to be produced.
+
+Extra Arguments
+===============
+
+`args` and `kwargs` are passed to the plotting function `plot_fn`.
+
+`MAX_NUM_COLS` and `MAX_NUM_ROWS` define the grid layout.
+"""
 function make_plots_generic(
-    output_path,
+    output_path::Union{<:AbstractString, Vector{<:AbstractString}},
     vars,
     args...;
     plot_fn = nothing,
     output_name = "summary",
     summary_files = String[],
     MAX_NUM_COLS = 1,
-    MAX_NUM_ROWS = 4,
+    MAX_NUM_ROWS = min(4, length(vars)),
     kwargs...,
 )
+
+    # We treat a Vector of 1 element as a normal string
+    length(output_path) == 1 && (output_path = output_path[1])
```

Comment:
```suggestion
    if output_path isa Vector && length(output_path) == 1
        output_path = output_path[1]
    end
```

Reply:
Actually, the logic is a bit complicated because `output_path` can be multiple things. I would suggest either:
 - Writing this exclusively for a vector `output_path`, or precomputing `vec_len_is_1 = output_path isa Vector && length(output_path) == 1` and using this for the special casing logic (and index into `output_path` on the fly)

Reply:
Maybe there's not too many edge cases. Just watch out for/fix when `length(output_path::String) == 1`, resulting in a `Char`.

Reply:
I tried cleaning up. I agree that it is not the best, but hopefully now it is more clear what's happening.

Reply:
@charleskawczynski any outstanding issues with this PR?

Reply:
No, I think it looks good. I just added it to the merge queue